### PR TITLE
feat: update keep_shutter_open and autofocus axis checkboxes

### LINKED ISSUE #230: fix: keep shutter open and autofocus axis
@tlambert03 what do you think if we remove the `keep_shutter_open` checkboxes from the `TimePlanWidget` and `ZPlanWidget` and move them into the general `MDAWidget` as in the video below?

About the autofocus axis, what do you think about having checkboxes in the general `MDAWidget` to select the axis and then using the behavior in the video below?

that's the WIP branch if you want to have a look https://github.com/fdrgsp/pymmcore-widgets/tree/shutter_and_af_axis

---
https://github.com/pymmcore-plus/pymmcore-widgets/assets/70725613/70748aab-b2c0-4f31-bf84-042a8b05a42f

### PR DESCRIPTION
**Pull request by fdrgsp**
related to #230 

- remove the `keep_shutter_open` checkboxes from the `TimePlanWidget` and `ZPlanWidget` and move them into the general `MDAWidget`, this way is also used in position sub-sequences.
- update the autofocus behavior by having p, t, g checkboxes in the general `MDAWidget` to select the axis that should be use to execute the autofocus during MDA acquisition.

Screenshot 2023-10-18 at 1 09 35 PM

closes #230

### CODE REVIEW

**Comment by tlambert03** [comment]
ready for review?

**Comment by fdrgsp** [comment]
> ready for review?

@tlambert03 Yes, I think you can have a look and see what you think. I also wanted to add a test for the QMessageBox I've added but I didn't manage yet (I'm also not sure if you like having the messagebox here or if you think is not necessary :))

**Comment by tlambert03** [comment]
I made some minor changes.  Mostly good.  I'm _slightly_ concerned with how logic-heavy/complicated things are getting.  But then again, I think I agree with all the aims here, so I guess it's better to have it :)

File: src/pymmcore_widgets/mda/_core_positions.py [hunk 1]
```diff
@@ -204,10 +207,18 @@ def _on_include_z_toggled(self, checked: bool) -> None:
         self.table().setColumnHidden(z_btn_col, not checked)
         super()._on_include_z_toggled(checked)
 
-    def _on_use_af_toggled(self, checked: bool) -> None:
+    def _on_af_per_position_toggled(self, checked: bool) -> None:
+        if checked and not self._mmc.isContinuousFocusLocked():
+            warnings.warn(
+                (
+                    f"The '{self._mmc.getAutoFocusDevice()}' Autofocus Device is NOT "
```
[the remark this answers is not in the document]
Comment (fdrgsp) [reply src/pymmcore_widgets/mda/_core_positions.py:214]: I agree! I think I'll remove it

File: src/pymmcore_widgets/useq_widgets/_mda_sequence.py [hunk 2]
```diff
@@ -222,6 +308,7 @@ def __init__(
         layout = QVBoxLayout(self)
         layout.addLayout(top_row)
         layout.addWidget(self.tab_wdg)
+        layout.addLayout(cbox_row)
```
[the remark this answers is not in the document]
Comment (fdrgsp) [reply src/pymmcore_widgets/useq_widgets/_mda_sequence.py:311]: @tlambert03 do you like it better?

Screenshot 2023-10-27 at 9 52 11 AM

**Pull request merged**